### PR TITLE
Add application gateway backendHealthOnDemand Operation

### PR DESCRIPTION
**Pull request by patoda**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json [hunk 1]
```diff
@@ -452,6 +452,70 @@
         "x-ms-long-running-operation": true
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/applicationGateways/{applicationGatewayName}/backendHealthOnDemand": {
```
Comment (KrisBash) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json]: POST action names should start with a verb, such as: getBackendHealth or listBackendHealth

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json [hunk 2]
```diff
@@ -692,6 +756,65 @@
...
+        "path": {
+          "type": "string",
+          "description": "Relative path of probe. Valid path starts from '/'. Probe is sent to <Protocol>://<host>:<port><path>"
+        },
+        "timeout": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The probe timeout in seconds. Probe marked as failed if valid response is not received with this timeout period. Acceptable values are from 1 second to 86400 seconds."
+        },
+        "pickHostNameFromBackendHttpSettings": {
```
Comment (KrisBash) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json]: As a general rule, a string enum is more future proof, either by modeling as enum with values "true" or "false" or refactoring to be: hostNameSourc with enum "backendHTTPSettings" or "default"

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see comment on POST action api name

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json (hunk 1 above)
Comment (patoda) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json]: changed to getBackendHealthOnDEmand

File: specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json (hunk 2 above)
Comment (patoda) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-02-01/applicationGateway.json]: would like to it keep same as an exisiting definition (ApplicationGatewayProbePropertiesFormat (line 1602)) for reuse

**Review by anton-evseev — CHANGES_REQUESTED** [review]

**Review by KrisBash — APPROVED** [review]
LGTM, thanks

**Comment by patoda** [comment]
@sergey-shandar , the backend NRP changes aren't in prod yet. Should we hold off this merge until the NRP changes are in prod. Sorry for confusion. I was under the impression that author is supposed to merge when approvals. Can we please revert and add DoNotMerge label until the NRP changes are in prod?

**Pull request merged**